release/build_pip_package.sh

```diff
@@ -16,44 +16,58 @@
 set -e
 set -x
 
+# Pick the Python that TFQ/TensorFlow used during configure/build.
+# Order: explicit env -> 3.11 -> python3
+PY="${PYTHON_BIN_PATH:-}"
+if [[ -z "$PY" ]]; then
+  if command -v python3.11 >/dev/null 2>&1; then
+    PY="$(command -v python3.11)"
+  elif command -v python3 >/dev/null 2>&1; then
+    PY="$(command -v python3)"
+  else
+    echo "ERROR: No suitable python found. Set PYTHON_BIN_PATH." >&2
+    exit 2
+  fi
+fi
+echo "Using Python: $PY"
+
+# Ensure packaging tools are present in THIS interpreter
+"$PY" - <<'PY' || { "$PY" -m pip install --upgrade pip setuptools wheel; }
+import importlib
+for m in ["setuptools","wheel"]:
+    importlib.import_module(m)
+PY
```

Comment:
This is an unconventional way of testing that certain packages have been installed. Is it not possible to use `pip show` and test its return value? For example, something like

```shell
if ! "$PY" -m pip show -q setuptools wheel > /dev/null 2>&1; then
  "$PY" -m pip install --upgrade pip setuptools wheel
fi
```